django/db/models/constraints.py

```diff
@@ -658,14 +658,21 @@ def validate(self, model, instance, exclude=None, using=DEFAULT_DB_ALIAS):
                         self.get_violation_error_message(),
                         code=self.violation_error_code,
                     )
-                # When fields are defined, use the unique_error_message() for
-                # backward compatibility.
-                for model, constraints in instance.get_constraints():
-                    for constraint in constraints:
-                        if constraint is self:
-                            raise ValidationError(
-                                instance.unique_error_message(model, self.fields),
-                            )
+                if self.violation_error_message == self.default_violation_error_message:
+                    # When fields are defined, use the unique_error_message() as
+                    # a default for backward compatibility.
+                    validation_error_message = instance.unique_error_message(
+                        model, self.fields
+                    )
+                    violation_error_code = validation_error_message.code
```

Comment:
You made the update as stated in [the comment](https://github.com/django/django/pull/17723#discussion_r1622019338) but it is missing the test to cover it (i.e. if I revert, the tests still pass)

Reply:
This is because the self.default_vaiolation_error_message has code = None. So it is literally replacing None by None. Not sure how to create a test for this.